### PR TITLE
feat(system-backup): volume backup policy

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/5011

### CODE REVIEW

**Comment by c3y1huang** [comment]
Test result: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/3915/

File: controller/system_backup_controller.go [hunk 1]
```diff
@@ -683,6 +700,151 @@ func (c *SystemBackupController) GenerateSystemBackup(systemBackup *longhorn.Sys
...
+
+func (c *SystemBackupController) backupVolumesIfNotPresent(systemBackup *longhorn.SystemBackup) (map[string]*longhorn.Backup, error) {
+	volumes, err := c.ds.ListVolumesRO()
+	if err != nil {
+		return nil, err
+	}
+
+	volumeBackups := make(map[string]*longhorn.Backup, len(volumes))
+	for _, volume := range volumes {
+		if volume.Status.LastBackup != "" {
```
Comment (shuo-wu) [inline controller/system_backup_controller.go:740]: Will this field be unset when the last backup is removed?

**Review by shuo-wu — COMMENTED** [review]
In general LGTM

File: controller/system_backup_controller.go (hunk 1 above)
Comment (innobead) [reply controller/system_backup_controller.go:740]: https://github.com/longhorn/longhorn-manager/blob/023dd3cbd5009f390608ea9cdd9f31bd397cf8a2/controller/volume_controller.go#L4270-L4271?

**Pull request merged**